### DISCUSSION TITLE
How to set different security level for introspection _service data type

### DISCUSSION THREAD

**Discussion by hrdoujak** [Netflix/dgs-framework#1625] [Q&A]
DGS framework implements resolvers for enhanced introspection query for `_service` type, which is needed by Apollo federation gateway: https://www.apollographql.com/docs/federation/subgraph-spec/#enhanced-introspection-with-query_service

For example:
```
{
  _service {
    sdl
  }
}
```

Is possible set special security level for this query? I would like to allow unsecured requests for this enhanced introspection query and OAuth for any other request to `/graphql` endpoint. 

I was looking for place/class where introspection resolvers is implemented to get some next hints, but unfortunatelly I did not find it.

Next solution can be use two endpoints; `/graphql` for all requests except introspection and `/introspection` just for introspection and than set security on endpoint level. Is that possible?

Thank you for any help.
Jakub Hrdoun

**Comment by srinivasankavitha** [comment]
There is a property :  dgs.graphql.introspection.enabled
https://netflix.github.io/dgs/configuration/#core-properties

You can turn this off for your service or use spring properties to control
which environment you want to turn it on/off for.

Hope that helps.
